feat: fixing login support on windows

rshade opened the pull request:
The fix addresses a Windows path issue in the Pulumi file:// URL construction. On Windows, file paths use backslashes, but URLs require forward slashes. The error you saw was occurring because
  backslashes in the Windows path were creating an invalid URL format.

  The modification:
  1. Creates a backendUrl variable
  2. Replaces all backslashes with forward slashes in the URL
  3. Uses the properly formatted URL for the backend

VenelinMartinov commented on pulumitest/newStack.go:
```diff
@@ -44,8 +44,11 @@ func (pt *PulumiTest) NewStack(t PT, stackName string, opts ...optnewstack.NewSt
 
 	if !options.UseAmbientBackend {
 		backendFolder := tempDirWithoutCleanupOnFailedTest(t, "backendDir", options.TempDir)
```
Should this return a linux path on windows in the first place?

rshade commented:
This also will fix #138

rshade commented:
```
 go test -count=1 -cover -timeout 2h -tags=nodejs -parallel 1 . -v 10
                                                                                                                                                                                                                                              Creating Checkpoint                                                                                                                                                                                                                             50%                                                                                                                                                                                                                                          [oooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo                                                                                                                 ]                                                                                                                                                                                                                                                       examples_nodejs_test.go:27: creating stack test
    examples_nodejs_test.go:27: linking yarn package: C:\Program Files (x86)\Yarn\bin\yarn.cmd link @pulumi/hyperv
    examples_nodejs_test.go:37: deploying
    examples_nodejs_test.go:39: Deployed resources successfully
    examples_nodejs_test.go:41: previewing update
    examples_nodejs_test.go:42: refreshing
    examples_nodejs_test.go:27: destroying stack, to skip this set PULUMITEST_SKIP_DESTROY_ON_FAILURE=true
    examples_nodejs_test.go:27: Removed temp directories. To retain these, set PULUMITEST_RETAIN_FILES=true
=== RUN   TestTsExamples/TestSimpleAllFour
    examples_nodejs_test.go:27: installing packages and plugins
    examples_nodejs_test.go:27: PULUMI_BACKEND_URL=file://C:/Users/rshade/AppData/Local/Temp/TestTsExamplesTestSimpleAllFour3340787826/002
    examples_nodejs_test.go:27: PULUMI_DEBUG_GRPC=C:\Users\rshade\AppData\Local\Temp\TestTsExamplesTestSimpleAllFour3340787826\003\grpc.json
    examples_nodejs_test.go:27: creating stack test
    examples_nodejs_test.go:27: linking yarn package: C:\Program Files (x86)\Yarn\bin\yarn.cmd link @pulumi/hyperv
    examples_nodejs_test.go:37: deploying
    examples_nodejs_test.go:39: Deployed resources successfully
    examples_nodejs_test.go:41: previewing update
    examples_nodejs_test.go:42: refreshing
    examples_nodejs_test.go:27: destroying stack, to skip this set PULUMITEST_SKIP_DESTROY_ON_FAILURE=true
    examples_nodejs_test.go:27: Removed temp directories. To retain these, set PULUMITEST_RETAIN_FILES=true
--- PASS: TestTsExamples (78.72s)
    --- PASS: TestTsExamples/TestDevEnv (39.19s)
    --- PASS: TestTsExamples/TestSimpleAllFour (39.53s)```

t0yv0 reviewed (approved):
The changes make sense to me.
